tasks/base/config/appname.go

```diff
@@ -107,43 +129,75 @@ func (t BaseConfigAppName) Execute(options tasks.Options, upstream map[string]ta
 
 	return tasks.Result{
 		Status:  tasks.Success,
-		Summary: fmt.Sprintf("%s unique application name(s) found.", strconv.Itoa(len(appNameInfoFromConfig))),
-		Payload: appNameInfoFromConfig,
+		Summary: fmt.Sprintf("%s unique application name(s) found.", strconv.Itoa(len(appNameInfosFromConfig))),
+		Payload: appNameInfosFromConfig,
 	}
 }
 
-func getAppNamesFromConfig(configElements []ValidateElement, configNames []string) []AppNameInfo {
+func getAppNameFromEnvVar(upstream map[string]tasks.Result) AppNameInfo {
+	envVars, ok := upstream["Base/Env/CollectEnvVars"].Payload.(map[string]string)
+
+	if !ok {
+		logger.Debug("Task did not meet requirements necessary to run: type assertion failure")
+	}
+	appname, isPresent := envVars[appNameEnvVarKey]
+	if !isPresent {
+		return AppNameInfo{}
+	}
+	return AppNameInfo{
+		Name:     appname,
+		FilePath: appNameEnvVarKey,
+	}
+}
+
+func getAppNamesFromConfig(configElements []ValidateElement) []AppNameInfo {
+
 	result := []AppNameInfo{}
 
-	for _, nameKey := range configNames {
-		for _, configFile := range configElements {
-
-			foundKeys := configFile.ParsedResult.FindKey(nameKey)
-			configFilePath := configFile.Config.FilePath
-			configFileName := configFile.Config.FileName
-
-			for _, key := range foundKeys {
-
-				if !key.IsLeaf() {
-					for _, child := range key.Children {
-						appName := child.Value()
-						result = append(result, AppNameInfo{
-							Name:     appName, // should we sanitize this?
-							FilePath: fmt.Sprintf("%s%s", configFilePath, configFileName),
-						})
-					}
-				} else {
-					appName := key.Value()
-
-					if len(appName) > 0 {
-						result = append(result, AppNameInfo{
-							Name:     appName, // should we sanitize this?
-							FilePath: fmt.Sprintf("%s%s", configFilePath, configFileName),
-						})
-					}
+	for _, configFile := range configElements {
+		foundKeys := findNameKeyInConfigFile(configFile)
+		configFilePath := configFile.Config.FilePath
+		configFileName := configFile.Config.FileName
+		/*
+			Only grab the first appname key found as this is the main required for an app to start reporting. The other appname keys are optional. Example:
+				/common/app_name: Luces-sqs-java
+				/development/app_name: My Application (Development)
+				/production/app_name: My Application (Production)
+				/staging/app_name: My Application (Staging)
+				/test/app_name: My Application (Test)
+		*/
+		if len(foundKeys) > 0 {
+			key := foundKeys[0]
+			if !key.IsLeaf() {
+				for _, child := range key.Children {
+					appName := child.Value()
+					result = append(result, AppNameInfo{
+						Name:     appName, // should we sanitize this?
```

Comment:
Can probably drop this comment here and on 184. Seems to be an artifact from orig. implementation.